AppController/djinn.rb

```diff
@@ -1151,26 +1151,28 @@ def update_node_info_cache
       @nodes.each { |node| ips << node.private_ip }
     }
     ips.each { |ip|
-      if ip == my_node.private_ip
-        begin
-          new_stats << JSON.load(get_node_stats_json(@@secret))
-        rescue FailedNodeException => exception
-          Djinn.log_warn("Failed to get local status update because: " \
-            "#{exception.message}")
-        end
-      else
-        threads << Thread.new {
+      threads << Thread.new {
+        Thread.current[:new_stat] = nil
+        if ip == my_node.private_ip
+          ret = get_node_stats_json(@@secret)
+        else
           acc = AppControllerClient.new(ip, @@secret)
+          ret = acc.get_node_stats_json
+        end
+        if ret == BAD_SECRET_MSG
```

Comment:
What do you think about putting this check in `AppControllerClient.make_call` and raising an exception there? That way, every place that uses the controller client library does not need to perform an additional check.

Reply:
good idea: let me know if this is what you had in mind